src/expressions/operator-expr.md

```diff
@@ -440,6 +440,28 @@ Casts an enum to its discriminant, then uses a numeric cast if needed.
 
 Casts to the `char` with the corresponding code point.
 
+#### Pointer to pointer cast
+
+Casting from a pointer to unsized type to pointer to sized type discards the pointer metadata, resulting in just a pointer to the referenced memory.
+Casting between pointers to unsized type preserves the pointer metadata unchanged (e.g. the slice element length remains the same).
```

Comment:
Should this say that the cast will only be allowed if both unsized types have the same "kind" of metadata (slice length / vtable)?

Reply:
Added in [8b2ea7c](https://github.com/rust-lang/reference/pull/1206/commits/8b2ea7cd7314c25d193fdc9d82aa6ae159bc0db5)